src/fenicsxconcrete/boundary_conditions/boundary.py

```diff
@@ -62,7 +60,7 @@ def boundary(x):
     return boundary
 
 
-def line_at(coordinates : list, dims : list):
+def line_at(coordinates: list[float], dims: list[str]) -> typing.Callable:
```

Comment:
```suggestion
def line_at(coordinates: list[float], dims: list[str, int]) -> typing.Callable:
```

Reply:
don't know about the body of the function, but as a user I would expect that `dims` could also be a list of int

Reply:
If that's the case, I add it would be list[str | int] then.